src/plugins/Network.js

```diff
@@ -47,28 +54,69 @@ module.exports = class Network extends OverlayPlugin {
          *            down_pkt_loss:<enabled/disabled>:<value>
          *            dns_delay:<enabled/disabled>:<value>
          */
-        this.instance.registerEventCallback('network_profile', (message) => {
-            const values = message.split(' ');
-            if (values.length < 9) {
+        this.instance.registerEventCallback('network_profile', this.handleNetworkProfile.bind(this));
+    }
+
+    handleSettings(message) {
+        const values = message.split(' ');
+
+        if (values[0] === 'if' ) {
+            this.wifiInput.disabled = false;
+            this.mobileInput.disabled = false;
+
+            if(values.length !== 3) {
                 return;
             }
-            const upSpeed = values[2].split(':');
-            const downSpeed = values[3].split(':');
-            const upDelay = values[4].split(':');
-            const downDelay = values[5].split(':');
-            const upPacketLoss = values[6].split(':');
-            const downPacketLoss = values[7].split(':');
-            const dnsDelay = values[8].split(':');
-
-            const isThrottlingEnabled = upSpeed[1] === 'enabled'
-                && downSpeed[1] === 'enabled'
-                && upDelay[1] === 'enabled'
-                && downDelay[1] === 'enabled'
-                && upPacketLoss[1] === 'enabled'
-                && downPacketLoss[1] === 'enabled'
-                && dnsDelay[1] === 'enabled';
-
-            const profile = PROFILES.find((elem) => {
+
+            const wifiOn = values[1].match(/(wifi:)(\w+)/);
+            if (wifiOn) {
+                this.wifiInputStatus = wifiOn[2] === 'on';
+            }
+            const mobileOn = values[2].match(/(mobile:)(\w+)/);
+            if (mobileOn) {
+                this.mobileInputStatus = mobileOn[2] === 'on';
+            }
+
+            if(this.widgedRendered) {
+                this.wifiInput.checked = this.wifiInputStatus;
+                this.mobileInput.checked = this.mobileInputStatus;
+                this.updateMobileSectionStatus();
+            }
+        } else if (values[0] === 'parameter' && values.length > 2 && values[2].includes("android_version")) {
+            const version = values[2].match(/(android_version:)(\w+)/);
+            this.androidVersion = version[2];
+            if (!this.widgedRendered) {
+                this.renderWidget();
+            }
+        }
+    }
+
+    handleNetworkProfile(message) {
```

Comment:
I found the comment in the Sim class, with the sample format of the messages, to be quite helpful.  :pray: Would it be possible to have a similar comment here? It might make it easier to understand how the message is being parsed.

Similar for `handleSettings`.